firebase-firestore/src/main/java/com/google/firebase/firestore/local/MemoryDocumentOverlayCache.java

```diff
@@ -50,11 +50,7 @@ public Overlay getOverlay(DocumentKey key) {
     return result;
   }
 
-  private void saveOverlay(int largestBatchId, @Nullable Mutation mutation) {
-    if (mutation == null) {
-      return;
-    }
-
```

Comment:
we should assert mutation != null, since `mutation.getKey()` is being used below.

Reply:
or does removing the `@Nullable` achieve the same thing?

Reply:
I added an explicit null check in `saveOverlays()`. @Nullable is just a compile-time check and doesn't affect runtime behavior. I also added a unit test.